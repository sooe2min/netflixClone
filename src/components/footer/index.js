import React from 'react'
import { Container, Title, Row, Column, Link, Select, Text, Copy, Break } from './styles/footer'

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Footer.Title = function FooterTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Footer.Row = function FooterRow({ children, ...restProps }) {
  return <Row {...restProps}>{children}</Row>
}

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>
}

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Footer.Select = function FooterSelect({ children, ...restProps }) {
  return <Select {...restProps}>{children}</Select>
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Footer.Copy = function FooterCopy({ children, ...restProps }) {
  return <Copy {...restProps}>{children}</Copy>
}

Footer.Break = function FooterBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>
}
