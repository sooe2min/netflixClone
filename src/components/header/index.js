import React, { useState } from 'react'
import { Background, Container, Logo, ButtonLink, Group, Link, Profile, Search, SearchIcon, SearchInput, Picture, Dropdown, Feature, FeatureCallOut, Text, PlayButton } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTurm, setSearchTurm, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src='/images/icons/search.png' alt='search' />
      </SearchIcon>
      <SearchInput
        placeholder='Search films and series'
        value={searchTurm}
        onChange={({ target }) => setSearchTurm(target.value)}
        active={searchActive}
      />
    </Search >
  )
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}


Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
