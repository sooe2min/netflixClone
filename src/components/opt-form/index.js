import React from 'react'
import { Container, Text, Input, Button, Break } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      {children}
    </Container>
  )
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

OptForm.Input = function OptFromInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Get Started' />
    </Button>
  )
}

OptForm.Break = function OptFormBreak({ children, ...restProps }) {
  return <Break {...restProps} />
}