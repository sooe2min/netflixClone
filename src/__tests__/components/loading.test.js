import React from 'react'
import { render } from '@testing-library/react'
import { Loading } from '../../components'

describe('<Loading />', () => {
  it('renders the <Loading /> with populated data', () => {
    const { container, getByTestId, getByRole } = render(
      <Loading src="/images/soomin.png" data-testid="loading" />
    )

    expect(getByTestId('loading')).toBeTruthy()
    expect(getByTestId('loading-picture')).toBeTruthy()
    expect(getByRole('img')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the <ReleaseBody />', () => {
    const { container } = render(
      <Loading.ReleaseBody data-testid="loading" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})