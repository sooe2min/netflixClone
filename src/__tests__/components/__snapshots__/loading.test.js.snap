// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders the <Loading /> with populated data 1`] = `
<div
  class="loading__Spinner-sc-11yhc25-1 kCevId"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-11yhc25-0 iYWgjc"
    data-testid="loading-picture"
    src="/images/users//images/soomin.png.png"
  />
</div>
`;

exports[`<Loading /> renders the <ReleaseBody /> 1`] = `null`;
